File: cases/taint/python/return-relay-two-hop-negative/return_relay.py
def dfb_source():  # DFB-SOURCE: return-two-hop-negative-input
    return "tainted"


def first_relay(value):  # DFB-WITNESS: return-two-hop-negative-first
    return value


def second_relay(value):  # DFB-WITNESS: return-two-hop-negative-second
    return first_relay(value)


def dfb_sink(value):  # DFB-SINK: return-two-hop-negative-sink
    pass


def run():
    result = second_relay(dfb_source())
    dfb_sink("clean")

File: cases/taint/ruby/return-relay-two-hop-negative/return_relay.rb
def dfb_source # DFB-SOURCE: return-two-hop-negative-input
  "tainted"
end

def first_relay(value) # DFB-WITNESS: return-two-hop-negative-first
  value
end

def second_relay(value) # DFB-WITNESS: return-two-hop-negative-second
  first_relay(value)
end

def dfb_sink(value) # DFB-SINK: return-two-hop-negative-sink
end

def run
  result = second_relay(dfb_source)
  dfb_sink("clean")
end
